aries_cloudagent/connections/models/diddoc/diddoc.py

```diff
@@ -56,8 +61,30 @@ def __init__(self, did: str = None) -> None:
         """
 
         self._did = canon_did(did) if did else None  # allow specification post-hoc
-        self._pubkey = {}
-        self._service = {}
+        fields = did.split(':')
+        if len(fields) > 2:
+            self._method = fields[1]
+            if len(fields) > 3:
+                self._network = fields[2]
+            else:
+                self._network = None
+        else:
+            self._method = 'sov'
+            self._network = None
+        self._pubkey = []
+        self._service = []
+
+    @property
+    def method(self) -> str:
+        """Accessor for DID method."""
+
+        return self._method
+
+    @property
+    def network(self) -> str:
+        """Accessor for DID network."""
+
+        return self._network
```

Comment:
DID Docs should not be aware of specific networks or methods of the DID. When creating references from DIDs, only the whole DID value is needed and should be given in its entirety to the DID Doc. There will be no need to put the DID together from different pieces.